lib/std/os/uefi/protocol/file.zig

```diff
@@ -40,108 +94,195 @@ pub const File = extern struct {
+        try self.setPosition(pos);
     }
 
-    pub fn getInfo(self: *const File, information_type: *align(8) const Guid, buffer_size: *usize, buffer: [*]u8) Status {
-        return self._get_info(self, information_type, buffer_size, buffer);
+    pub fn getInfo(
+        self: *const File,
+        information_type: *align(8) const Guid,
+        buffer: []u8,
+    ) GetInfoError![]u8 {
+        var len = buffer.len;
+        switch (self._get_info(self, information_type, &len, buffer.ptr)) {
+            .success => return buffer[0..len],
+            .unsupported => return Error.Unsupported,
+            .no_media => return Error.NoMedia,
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .buffer_too_small => return Error.BufferTooSmall,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
     }
 
-    pub fn setInfo(self: *const File, information_type: *align(8) const Guid, buffer_size: usize, buffer: [*]const u8) Status {
-        return self._set_info(self, information_type, buffer_size, buffer);
+    pub fn setInfo(
+        self: *const File,
+        information_type: *align(8) const Guid,
+        buffer: []const u8,
+    ) SetInfoError!void {
+        switch (self._set_info(self, information_type, buffer.len, buffer.ptr)) {
+            .success => {},
+            .unsupported => return Error.Unsupported,
+            .no_media => return Error.NoMedia,
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .write_protected => return Error.WriteProtected,
+            .access_denied => return Error.AccessDenied,
+            .volume_full => return Error.VolumeFull,
+            .bad_buffer_size => return Error.BadBufferSize,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
     }
 
-    pub fn flush(self: *const File) Status {
-        return self._flush(self);
+    pub fn flush(self: *File) FlushError!void {
+        switch (self._flush(self)) {
+            .success => {},
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .write_protected => return Error.WriteProtected,
+            .access_denied => return Error.AccessDenied,
+            .volume_full => return Error.VolumeFull,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
     }
 
-    pub const efi_file_mode_read: u64 = 0x0000000000000001;
-    pub const efi_file_mode_write: u64 = 0x0000000000000002;
-    pub const efi_file_mode_create: u64 = 0x8000000000000000;
+    pub const OpenMode = enum(u64) {
+        read = 0x0000000000000001,
+        // implies read
+        write = 0x0000000000000002,
+        // implies read+write
+        create = 0x8000000000000000,
+    };
+
+    pub const Attributes = packed struct(u64) {
```

Comment:
These are duplicated in `std/os/uefi.zig`, maybe we can use this as the source of truth